File: Solution3136.cpp
/*
Runtime 0ms, Beats 100.00%
Memory 7.84MB, Beats 86.98%

submitted at Jul 16, 2025 22:17
*/


class Solution {
public:
    bool isValid(string word) {
        if (word.length() < 3) return false;

        int vowel = 0;
        int consonant = 0;
        int digit = 0;
        for (int i = 0; i < word.length(); i++) {
            char c = word[i];
            if (c >= 'A' && c <= 'Z') c = c - 'A' + 'a';

            if (c >= 'a' && c <= 'z') {
                switch (c) {
                    case 'a':
                    case 'e':
                    case 'i':
                    case 'o':
                    case 'u':
                        vowel += 1;
                        break;
                    default:
                        consonant += 1;
                }
                continue;
            }

            if (c >= '0' && c <= '9') {
                digit += 1;
                continue;
            }

            return false;
        }
        if(vowel > 0 && consonant > 0)
            return true;
        return false;
    }
};

File: Rust/solution3136.rs
impl Solution {
    pub fn is_valid(word: String) -> bool {
        if word.len() < 3 {
            return false;
        }

        let mut vowel = 0;
        let mut consonant = 0;
        let mut digit = 0;

        for mut c in word.chars() {
            // Convert uppercase to lowercase
            if c.is_ascii_uppercase() {
                c = c.to_ascii_lowercase();
            }

            if c.is_ascii_lowercase() {
                match c {
                    'a' | 'e' | 'i' | 'o' | 'u' => vowel += 1,
                    _ => consonant += 1,
                }
                continue;
            }

            if c.is_ascii_digit() {
                digit += 1;
                continue;
            }

            // If any other character (non-alphanumeric)
            return false;
        }

        vowel > 0 && consonant > 0
    }
}